core/src/main/java/org/stellar/anchor/util/NetUtil.java

```diff
@@ -17,7 +17,14 @@ public static String fetch(String url) throws IOException {
     Request request = OkHttpUtil.buildGetRequest(url);
     Response response = getCall(request).execute();
 
-    if (response.body() == null) return "";
+    // Check if response was successful (status code 200)
+    if (!response.isSuccessful()) {
+      throw new IOException(String.format("Unable to fetch data from %s", url));
+    }
```

Comment:
Are we sure we want to throw an exception in this case? We aren't interested in the response body of a 400 transaction?

Reply:
previously we were just returning "".   I agree with you though. Will you be ok if i file a ticket for this and make a bug fix for this?